sentry_sdk/sessions.py

```diff
@@ -19,32 +20,56 @@
     from typing import Union
 
 
-def is_auto_session_tracking_enabled(hub=None):
-    # type: (Optional[sentry_sdk.Hub]) -> Union[Any, bool, None]
-    """Utility function to find out if session tracking is enabled."""
-    # TODO: add deprecation warning
-
-    if hub is None:
-        hub = sentry_sdk.Hub.current
-
-    should_track = hub.scope._force_auto_session_tracking
+def _is_auto_session_tracking_enabled(scope):
+    # type: (sentry_sdk.Scope) -> bool
+    """
+    Utility function to find out if session tracking is enabled.
+    """
 
+    should_track = scope._force_auto_session_tracking
     if should_track is None:
-        client_options = hub.client.options if hub.client else {}
+        client_options = sentry_sdk.get_client().options
         should_track = client_options.get("auto_session_tracking", False)
 
     return should_track
 
 
+@contextmanager
+def track_session(scope, session_mode="application"):
```

Comment:
I originally tried to make the `auto_session_tracking` function use the scope based API by default but still be backwards compatible with hub-based API, but implementing this was going to be extremely overcomplicated.

I decided it would be much simpler to just implement a new function to replace `auto_session_tracking` instead, hence why I have proposed fully deprecating `auto_session_tracking` and `auto_session_tracking_scope` and replacing both with the new `track_session`.